test/widgets/profile/edit_brith_date_page_test.dart

```diff
@@ -0,0 +1,140 @@
+      'user': User(
+          id: authId,
+          appMetadata: {},
+          userMetadata: {},
+          aud: 'public',
+          createdAt: DateTime.now().toIso8601String(),
+          email: email),
+      'email': email,
+    });
+    whenRequest(processor, urlMatcher: contains('/rest/v1/profiles')).thenReturnJson(profile.toJsonForApi());
+  });
+  group('edit_birth_date_page', () {
+    testWidgets('birthDate TextField', (WidgetTester tester) async {
+      //load page
+      await pumpMaterial(tester, EditBirthDatePage(profile));
+      await tester.pump();
+
+      //check if birthDate is displayed
+      expect(find.text(localeManager.formatDate(profile.birthDate!)), findsOneWidget);
+
+      //check if birthDate TextField is displayed
+      final Finder birthDateInput = find.byKey(const Key('birthDateInput'));
+      expect(birthDateInput, findsOneWidget);
+
+      //check if birthDate TextField is editable
+      await tester.tap(birthDateInput);
+      await tester.pumpAndSettle();
+      expect(find.byType(DatePickerDialog), findsOneWidget);
+    });
+    testWidgets('change birthDate', (WidgetTester tester) async {
+      //load page
+      await pumpMaterial(tester, EditBirthDatePage(profile));
+      await tester.pump();
+
+      //check if birthDate is displayed
+      final Finder birthDateInput = find.text(localeManager.formatDate(profile.birthDate!));
+      expect(birthDateInput, findsOneWidget);
+
+      //tap birthDate TextField
+      await tester.tap(find.byKey(const Key('birthDateInput')));
+      await tester.pumpAndSettle();
+
+      //tap in DatePickerDialog to custom input
+      await tester.tap(find.byIcon(Icons.edit));
+      await tester.pumpAndSettle();
+
+      //enter new birthDate
+      await tester.tap(birthDateInput.last);
+      final String date = localeManager.formatDate(profile.birthDate!.add(const Duration(days: 5)));
+
+      //leave DatePickerDialog
+      await tester.enterText(birthDateInput.last, date);
+      await tester.tap(find.text('OK'));
+      await tester.pumpAndSettle();
+
+      //check if new birthDate is displayed
+      expect(find.text(date), findsOneWidget);
+    });
+    testWidgets('clear Button', (WidgetTester tester) async {
+      //load page
+      await pumpMaterial(tester, EditBirthDatePage(profile));
+      await tester.pump();
+
+      //check if birthDate is displayed
+      expect(find.text(localeManager.formatDate(profile.birthDate!)), findsOneWidget);
+
+      //check if clear Button is displayed
+      final Finder clearButton = find.byKey(const Key('clearButton'));
+      expect(clearButton, findsOneWidget);
+
+      //check if birthDate is cleared
+      await tester.tap(clearButton);
+      await tester.pump();
+      expect(find.text(localeManager.formatDate(profile.birthDate!)), findsNothing);
+    });
+    testWidgets('save Button', (WidgetTester tester) async {
+      //sign in
+      SupabaseManager.supabaseClient.auth.signInWithPassword(
```

Comment:
let's put this in setUp, it's not immediately clear that this is specific to this one case